src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -743,6 +779,28 @@ protected void processFiltered(File file, List<String> lines) throws CheckstyleE
         }
     }
 
+    private static class DummyFileSetViolationCheck extends AbstractFileSetCheck
+        implements ExternalResourceHolder {
```

Comment:
please create UT base on AbstractCheck that file violation but it is suppressed by filter.
I do not like the idea to spread our hack of `fireErrors` in `finishProcessing` method.